test/integration/Test/Integration/Framework/DSL.hs

```
@@ -265,6 +269,18 @@ defaultDistribution
 defaultDistribution c s = pure $
     PaymentDistribution (WalAddress $ head $ s ^. typed) (WalletCoin $ mkCoin c)
 
+customDistribution
+    :: [Account]
+    -> [Word64]
+    -> NonEmpty PaymentDistribution
+customDistribution accounts coins =
+    let recepientWalAddresses = map (\acc -> acc ^. walAddresses)
+                                $ concatMap (\acc -> acc ^. addresses)
+                                $ accounts
+    in fromList $
+       zipWith PaymentDistribution recepientWalAddresses (map (\coin -> WalletCoin $ mkCoin coin) coins)
```

This may lead to unexpected results if:

- The `accounts` and `coins` list don't have the same number of elements
- The `accounts` have more than one address 

I'd recommend to either check that as invariant and fail hard if the invariant aren't met, or (better IMO), require the caller to provide the a `(Account, Word64)]` himself and only take the first address of each account -- failing if there's none maybe. Or asking directly for `[(Address, Word64)]`

---

done